crates/sui-sdk/examples/utils/example_utils.rs

```diff
@@ -0,0 +1,180 @@
+    }
+
+    let coins = client
+        .coin_read_api()
+        .get_coins_stream(active_address, Some(coin_type.clone()))
+        .filter(|coin| future::ready(coin.balance >= 100_000_000))
+        .collect::<Vec<_>>()
+        .await;
+    let coin_obj_id = coins.get(0).unwrap().coin_object_id;
+    return Ok(coin_obj_id);
+}
+
+/// Request tokens from the Faucet for the given address
+pub(crate) async fn request_tokens_from_faucet(address: SuiAddress) -> Result<(), anyhow::Error> {
+    let address_str = address.to_string();
+    let json_body = json![{
+        "FixedAmountRequest": {
+            "recipient": &address_str
+        }
+    }];
+
+    // make the request to the faucet JSON RPC API for coin
+    let client = Client::new();
+    let resp = client
+        .post(SUI_FAUCET)
+        .header("Content-Type", "application/json")
+        .json(&json_body)
+        .send()
+        .await?;
+    println!("Faucet request status: {}", resp.status());
+    // wait for a bit before jumping to other methods
+    std::thread::sleep(std::time::Duration::from_secs(2));
+    Ok(())
+}
+
+/// Return a transaction digest from a split coin + merge coins transaction
+pub(crate) async fn split_coin_digest(
+    sui: &SuiClient,
+    sender: &SuiAddress,
+) -> Result<TransactionDigest, anyhow::Error> {
+    let coin_type = "0x2::sui::SUI".to_string();
+    let coins = sui
+        .coin_read_api()
+        .get_coins(*sender, Some(coin_type), None, None)
+        .await?;
+
+    let coins = coins.data;
+    let coins_len = coins.len();
+    let coin_for_split = coins
+        .into_iter()
+        .filter(|x| x.balance >= 500_000)
+        .collect::<Vec<_>>();
+
+    let coin = coin_for_split
+        .get(0)
+        .expect("Tried to get a coin with enough SUI from sender: {sender}, but it does not exist even though it should. Aborting");
+
+    println!(
+        "Address {sender} has {} coins. The selected coin for split is {} and has a balance of {}\n", coins_len, coin.coin_object_id, coin.balance);
+
+    let max_gas_budget = 2_000_000;
+
+    let gas_coin = GasCoin::new(coin.coin_object_id, 10000000);
+    println!("{:?}", gas_coin);
+    let txb = sui.transaction_builder();
+    let txb_res = txb
+        .split_coin(
+            *sender,
+            *gas_coin.id(),
+            // coin.coin_object_id,
+            vec![10],
+            None,
+            // Some(*gas_coin.id()),
+            max_gas_budget,
+        )
+        .await?;
+    let tx_data: TransactionData = txb_res.into();
+    let new_coin = tx_data.execution_parts().2.get(0).unwrap().0;
+
+    println!(
+        "Merge back the new split coin to the original sender and the coin from which it was split"
+    );
+    let txb_res = txb
+        .merge_coins(
+            *sender,
+            coin.coin_object_id,
+            new_coin,
+            Some(*gas_coin.id()),
+            max_gas_budget,
+        )
+        .await?;
```

Comment:
I don't fully follow what's going on here, but it looks like you're trying to create a transaction that splits a coin from the gas coin and then merges it back, in one PTB.  But it looks like you're actually creating two separate `TransactionData`'s -- one for the split, which is never run, and one for the merge which you do run.

Here's how I'd achieve what you're trying to do (offered without compiling/running):

```rust
use sui_types::programmable_transaction_builder::ProgrammableTransactionBuilder;

let mut ptb = ProgrammableTransactionBuilder::new();
ptb.command(Command::SplitCoin(Argument::GasCoin, vec![1000]));
ptb.command(Command::MergeCoin(Argument::GasCoin, vec![Argument::Result(0)]);
let tx_data = TransactionData::new(
    TransactionKind::programmable(builder.finish()),
    *sender,
    coin.coin_object_id,
    max_gas_budget,
    sui.read_api.get_reference_gas_price().await.unwrap(),
);
```